### ISSUE TITLE
Add JSON to REAL support

### ISSUE THREAD

**Issue by wenleix** [prestodb/presto#7705]
```
presto:> select CAST(JSON '{"a":1.1}' AS MAP<VARCHAR, REAL>);
Query 20170330_212623_17439_9htcy failed: Cannot cast JSON to map(varchar,real)

presto:> select CAST(JSON '{"a":1.1}' AS MAP<VARCHAR, DOUBLE>);
  _col0
---------
 {a=1.1}
(1 row)
```

CAST JSON TO MAP support was added before `REAL` introduced in Presto. 

The current workaround is first CAST TO `MAP<DOUBLE>`, then CAST TO `MAP<REAL>`.

**Comment by wenleix** [comment]
Also, CAST JSON TO REAL is also not currently supported.

```
presto> select cast(json '0.2' as real);
Query 20170405_062119_11694_kmjsc failed: line 1:8: Cannot cast json to real
select cast(json '0.2' as real)

presto> select cast(json '0.2' as double);
 _col0
-------
   0.2
(1 row)
```

**Comment by wenleix** [comment]
Resolved by #7740
